### PR TITLE
Fix n+1 problem in PostsController#index

### PR DESCRIPTION
**Pull request by peter-akhigbe**
- Fix N+1 Problem
- Integration tests with Capybara

Before:

Screenshot 2023-12-22 at 8 31 56 AM
Screenshot 2023-12-22 at 9 15 39 AM

After:

Screenshot 2023-12-22 at 8 50 34 AM

### CODE REVIEW

**Comment by peter-akhigbe** [comment]
Hello @ reviewer

My partner had some issues, he could not join me.
so i implemented the complete project requirements alone.

IMG_3653

File: spec/views/users/users_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      click_link post.title
+
+      expect(current_path).to eq(user_post_path(user, post))
+    end
+
+    it 'redirects to the user show page to see all posts' do
+      visit user_path(user)
+      click_link('See all posts')
+
+      expect(current_path).to eq(users_path)
```
Comment (Ibizugbe) [inline spec/views/users/users_show_spec.rb:48]: For the `user_spec.rb` kindly add a test that asserts that you can see the user's first 3 posts, as this is a [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md) in this milestone😉😉

image

File: spec/views/posts/posts_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      expect(page).to have_content('Peter')
+      expect(page).to have_content('Number of post: 6')
+    end
+
+    it 'displays post information' do
+      expect(page).to have_content('Post Three')
+      expect(page).to have_content('three: post text')
+      expect(page).to have_content('comment one')
+      expect(page).to have_content('Likes: 0')
+    end
```
Comment (Ibizugbe) [inline spec/views/posts/posts_index_spec.rb:37]: Nice job so far👍🏼. As a [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md), kindly add a test that asserts that you can

- [ ] How many comments a post has😉.
- [ ] See a section for pagination if there are more posts than fit on the view😉.
image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @peter-akhigbe 👋🏼👋🏼👋🏼, 

request change


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] The required tests for the post_show page have been written👍🏼.
- [x] All tests are passing👍🏼.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — APPROVED** [review]
Hi @peter-akhigbe, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

APPRVED
## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**